# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/dns/nsIDNSService.idl
# Commit: af0f14eed70e
# Full Hash: af0f14eed70edb2abb1ecd3cae4928387287d3e1
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2024-06-14 20:49:45
# Regressor Bug: 1901990
# File Overlap Count: 2
# Description:
#   Bug 1901990 - Introduce a pref for setting a mock HTTPS RR, r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D213559
# ==============================================================================

diff -r 9f913cbd60ee -r af0f14eed70e netwerk/dns/nsIDNSService.idl
--- a/netwerk/dns/nsIDNSService.idl	Thu Jun 13 15:08:04 2024 +0200
+++ b/netwerk/dns/nsIDNSService.idl	Fri Jun 14 09:11:49 2024 +0000
@@ -93,9 +93,11 @@
         RESOLVE_WANT_RECORD_ON_ERROR = (1 << 16),
         // If set, the native HTTPS query is not allowed.
         RESOLVE_DISABLE_NATIVE_HTTPS_QUERY = (1 << 17),
+        // For testing purposes only. If set, create a mock HTTPS RR and use it.
+        RESOLVE_CREATE_MOCK_HTTPS_RR = (1 << 18),
 
         // Bitflag containing all possible flags.
-        ALL_DNSFLAGS_BITS = ((1 << 18) - 1),
+        ALL_DNSFLAGS_BITS = ((1 << 19) - 1),
     };
 
     cenum ConfirmationState : 8 {